deltachat-ios/AppDelegate.swift

```diff
@@ -484,35 +484,33 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
         pushToDebugArray("1")
 
         // move work to non-main thread to not block UI (otherwise, in case we get suspended, the app is blocked totally)
-        // (we are using `qos: default` as `qos: .background` or `main.asyncAfter` may be delayed by tens of minutes)
+        // (we are using `qos: default` as `qos: .background` may be delayed by tens of minutes)
         DispatchQueue.global().async { [weak self] in
             guard let self else { completionHandler?(.failed); return }
 
-            // we're in background, run IO for a little time
-            self.dcAccounts.startIo()
-            self.dcAccounts.maybeNetwork()
             self.pushToDebugArray("2")
 
             self.addDebugFetchTimestamp()
-
-            // create a new semaphore to make sure the received DC_CONNECTIVITY_CONNECTED really belongs to maybeNetwork() from above
-            // (maybeNetwork() sets connectivity to DC_CONNECTIVITY_CONNECTING, when fetch is done, we're back at DC_CONNECTIVITY_CONNECTED)
             self.dcAccounts.fetchSemaphore = DispatchSemaphore(value: 0)
+            let start = CFAbsoluteTimeGetCurrent()
+
+            // backgroundFetch() pauses IO as needed
+            if !self.dcAccounts.backgroundFetch(timeout: 20) {
+                logger.error("backgroundFetch failed")
+                self.pushToDebugArray("ERR2")
```

Comment:
What does `ERR2` stand for?

Reply:
it js just a random short  string added to "Settings/Advaned/View Log". it makes sense only if you look at the code. once things are settled, we can remove this DebugArray completely